### PR TITLE
Expose containers log paths on host

### PR DESCRIPTION
**Pull request by wendigo**
This helps to debug CI failures by keeping Hadoop/Presto logs on the host machine

### CODE REVIEW

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java [hunk 1]
```diff
@@ -82,8 +83,8 @@ public void run()
         @Option(name = "--environment", title = "environment", description = "the name of the environment to start", required = true)
         public String environment;
 
-        @Option(name = "--startup-timeout", title = "environment startup timeout", description = "environment startup timeout")
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java:85]: hmm. I recall we already remove it. Please make separate commit anyway.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java [hunk 2]
```diff
@@ -81,6 +91,21 @@ public void copyFileToContainer(Transferable transferable, String containerPath)
...
 
+    public DockerContainer withExposedLogPaths(String... logPaths)
+    {
+        this.logPaths.addAll(Arrays.asList(logPaths));
+        return this;
+    }
+
+    public void exposeLogsInHostPath(Path hostBasePath)
+    {
+        for (String containerLogPath : logPaths) {
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:102]: there can be quite a lot of those. Is there a limit on number of directories bound...
I guess not very low - we used to bind a lot of stuff before moving to copying.

**Review by losipiuk — APPROVED** [review]
Nice. Thank you.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java (hunk 2 above)
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:102]: Neither docker java nor docker docs itself mention any limit on the number of binds

**Comment by wendigo** [comment]
@losipiuk done :)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java [hunk 3]
```diff
@@ -82,8 +83,8 @@ public void run()
...
-        public Duration startupTimeout = Duration.valueOf("10m");
+        @Option(name = "--logs-dir", title = "logs dir", description = "location of the exported logs directory")
```
Comment (electrum) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java:86]: Please rebase onto #4908

`title` should become `paramLabel = `<dir>` and `description` should start with upppercase

Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java:86]: Sure. Thx @electrum :)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java [hunk 4]
```diff
@@ -97,8 +95,11 @@ public void run()
         @Option(name = "--environment", title = "environment", description = "the name of the environment to start", required = true)
         public String environment;
 
+        @Option(name = "--logs-dir", title = "logs dir", description = "location of the exported logs directory")
+        public String logsDirBase = "logs/";
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java:99]: If i eg invoke test run from intellij directly, where will the logs go to?
the logs/ folder does not exist, right?

maybe it should be null by default, or tempdir?

(same for env up)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java [hunk 5]
```diff
@@ -97,8 +95,11 @@ public void run()
...
+
         @Option(name = "--reports-dir", title = "reports dir", description = "location of the reports directory")
-        public String reportsDir = "presto-product-tests/target/";
+        public String reportsDir = "presto-product-tests/target/reports";
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java:102]: separate commit?

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java [hunk 6]
```diff
@@ -81,6 +91,21 @@ public void copyFileToContainer(Transferable transferable, String containerPath)
...
+            Path hostLogPath = Paths.get(hostBasePath.toString(), containerLogPath);
+            cleanOrCreateHostPath(hostLogPath);
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:104]: I'd rather see caller do the cleanup (if needed)

This configuration method should ideally be just that -- a configuration method without side effects

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java [hunk 7]
```diff
@@ -81,6 +91,21 @@ public void copyFileToContainer(Transferable transferable, String containerPath)
         copyFileToContainer(containerPath, () -> super.copyFileToContainer(transferable, containerPath));
     }
 
+    public DockerContainer withExposedLogPaths(String... logPaths)
+    {
+        this.logPaths.addAll(Arrays.asList(logPaths));
+        return this;
+    }
+
+    public void exposeLogsInHostPath(Path hostBasePath)
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:100]: `withExposedLogPaths` and `exposeLogsInHostPath` look like configuration methods, but currently need to be called in correct order

let's add some verify to ensure they are always callled in that order.
simplest way would be

- to null out `logPaths` inside `exposeLogsInHostPath`
- to check `logPaths != null` yet in `withExposedLogPaths`

(i don't see a need to support the methods being called in varying order, so let's not spend time on that)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java [hunk 8]
```diff
@@ -95,4 +120,24 @@ private static void verifyHostPath(String hostPath)
...
     }
+
+    public static void cleanOrCreateHostPath(Path path)
+    {
+        try {
+            if (Files.exists(path)) {
+                Files.walk(path)
+                        .sorted(reverseOrder())
+                        .map(Path::toFile)
+                        .forEach(File::delete);
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:131]: See https://github.com/prestosql/presto/blob/51263348564d2f1853aad0dbae30ab212fc858a9/presto-raptor-legacy/src/test/java/io/prestosql/plugin/raptor/legacy/storage/TestOrcFileRewriter.java#L81

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java (hunk 4 above)
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java:99]: to logs/ dir which is emptied on every run.

File: presto-product-tests-launcher/src/main/resources/launcher.properties [hunk 9]
```diff
@@ -0,0 +1 @@
+project.version=${project.version}
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/resources/launcher.properties:1]: where do we need this?

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentOptions.java [hunk 10]
```diff
@@ -33,35 +30,12 @@
     @Option(names = "--without-presto", description = "Do not start presto-master")
     public boolean withoutPrestoMaster;
 
-    @Option(names = "--bind", description = "Bind ports on localhost")
-    public boolean bindPorts = toBoolean(firstNonNull(System.getenv("PTL_BIND_PORTS"), "true"));
+    @Option(names = "--no-bind", description = "Bind ports on localhost", negatable = true)
+    public boolean bindPorts = true;
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentOptions.java:34]: What drive what is default value of option here. The value assigned to `bindPorts` variable?
If so can you just declare option ss `@Option(names = "--bind", description = "Bind ports on localhost", negatable = true)`

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/SuiteRun.java [hunk 11]
```diff
@@ -153,28 +154,41 @@ public void run()
...
+            }
+            else {
+                log.info("Suite %s succeeded in %s: ", suiteName, nanosSince(startTime));
+            }
+
+            results.stream()
+                    .filter(TestRunResult::hasFailed)
+                    .forEach(Execution::printTestRunSummary);
+
+            results.stream()
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/SuiteRun.java:180]: I am not convinced we need those. And even if we do,  I would put them before failed ones. So it is easy to scroll down to failed list of failed tests on GHA.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/SuiteRun.java [hunk 12]
```diff
@@ -216,11 +219,19 @@ public TestRunResult executeSuiteTestRun(String suiteName, SuiteTestRun suiteTes
...
+            testRunOptions.reportsDir = Paths.get("presto-product-tests/target/reports/" + suiteRunId);
+            testRunOptions.logsDirBase = logsDirBase.resolve(suiteRunId);
             return testRunOptions;
         }
+
+        private static String suiteRunId(String suiteName, SuiteTestRun suiteTestRun, EnvironmentConfig environmentConfig)
+        {
+            List<String> testArguments = suiteTestRun.getTemptoRunArguments(environmentConfig);
+            // suite-2 contains two singlenode environment test runs - use tempto arguments to uniquely identify each test run
+            String testArgumentsHash = Hashing.sha256().hashString(Joiner.on("-").skipNulls().join(testArguments), Charsets.UTF_8).toString();
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/SuiteRun.java:232]: this is not useful.
Can we make it a single digit number. Only present if there actually is a conflict. 
Or alternatively you can just put an index of `SuiteTestRun` in the list in the string. Right?

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentOptions.java (hunk 10 above)
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentOptions.java:34]: No I can't. It's just the way negatable works (so by default it's --bind, --no-bind will switch to false)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java (hunk 4 above)
Comment (findepi) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java]: At the very least it should be gitignored, but IMO (currently) not exporting logs by default is more reasonable.

Comment (losipiuk) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java]: Yeah - I think it makes sense that default for `--logs-dir` should be null. And we handle that in a way that we do not bind any host directories for logs to containers.
